src/libsyntax/ext/base.rs

```diff
@@ -776,6 +777,22 @@ impl<'a> ExtCtxt<'a> {
     pub fn name_of(&self, st: &str) -> ast::Name {
         token::intern(st)
     }
+
+    pub fn suggest_macro_name(&mut self, name: &str, span: Span) {
+        use std::cmp::max;
+
+        let mut min: Option<(Name, usize)> = None;
+        let max_dist = max(name.len() / 3, 1);
+        for macro_name in self.syntax_env.names.iter() {
+            let dist = lev_distance(name, &macro_name.as_str());
+            if dist <= max_dist && (min.is_none() || min.unwrap().1 > dist) {
+                min = Some((*macro_name, dist));
+            }
+        }
+        if let Some((suggestion, _)) = min {
+            self.span_help(span, &format!("did you mean `{}`?", suggestion));
```

Comment:
Use `fileline_help` please. Printing the span multiple times is not useful and only contributes to more useless visual noise.